libs/langchain-community/src/utils/tencent_hunyuan.ts

```diff
@@ -0,0 +1,56 @@
+import { createHash, createHmac, BinaryLike } from "crypto";
+
+const host = "hunyuan.tencentcloudapi.com";
```

Comment:
Should we make this configurable?

Reply:
> Should we make this configurable?

I made `host` configurable in the newest commit. Generally, users do not need to care or change the host, since hunyuan only has one `host` in Tencent Cloud, and is stable. However, consider the needs for debug in development/testing environment, making host configurable seems reasonable.

Reply:
> > This will not work in all environments - you'll need to import from `node:crypto` or [SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) depending on what environments you care about, then mark these entrypoints as `requiresOptionalDependencies` in `langchain.config.js`
> 
> I'd like to make this model usable either in node/browser environment. Is there some example that detectes the environment and import the appropriate package to use?

Problem (may) solved. 

The `crypto` package is used to sign the request for Tencent Cloud. In the newest commit, I provided two version of sign in `utils/tencent_hunyuan/index.ts` and `utils/tencent_hunyuan/web.ts`, with the former one using nodejs's `crypto` package and the latter one using `crypto-js` package (which supports browser environment). Users in different environment may import different version of `sign` method, and pass it to the model class. I updated documents and examples to illustrate this.

Do you think this is an acceptable way to suuport both nodejs and browser.

Reply:
Yes this is ok, it may be more intuitive to make `chat_models/tencent/web` and `chat_models/tencent` instead of this `utils` package. See:

https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain-community/src/chat_models/bedrock